src/main/java/com/academy/config/authentication/AuditorAwareBean.java

```diff
@@ -0,0 +1,39 @@
+package com.academy.config.authentication;
+
+import jakarta.servlet.http.HttpServletRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.AuditorAware;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+
+import java.util.Optional;
+
+@Component("auditorProvider")
+public class AuditorAwareBean implements AuditorAware<String> {
+
+    @Autowired(required = false)
+    private HttpServletRequest request;
+
+    @Override
+    public Optional<String> getCurrentAuditor() {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
```

Comment:
Aqui podes usar o AuthenticationFacade
```suggestion 
Authentication authentication = authenticationFacade.getAuthentication(); 
```